server/odc-service/src/main/java/com/oceanbase/odc/config/HookConfiguration.java

```diff
@@ -142,18 +144,22 @@ private void approvalFlowConfigUsageCheck(Long flowConfigId) {
     }
 
     private void projectReferenceCheck(Long userId, Long organizationId) {
+        Map<Long, ProjectEntity> id2Project =
+                projectRepository.findAllByOrganizationId(organizationId).stream().filter(p -> !p.getArchived())
+                        .collect(Collectors.toMap(ProjectEntity::getId, p -> p));
+
         Map<Long, Set<ResourceRoleName>> projectId2ResourceRoleNames =
                 resourceRoleService.getProjectId2ResourceRoleNames(userId).entrySet().stream()
                         .filter(e -> e.getValue().contains(ResourceRoleName.OWNER)
                                 || e.getValue().contains(ResourceRoleName.DBA))
+                        .filter(e -> id2Project.containsKey(e.getKey()))
                         .collect(Collectors.toMap(Map.Entry::getKey, Map.Entry::getValue));
         if (projectId2ResourceRoleNames.size() == 0) {
             return;
         }
-        Map<Long, ProjectEntity> id2Project =
-                projectRepository.findAllByOrganizationId(organizationId).stream().filter(p -> p.getArchived() == false)
-                        .collect(Collectors.toMap(ProjectEntity::getId, p -> p));
-        String names = projectId2ResourceRoleNames.keySet().stream().map(id -> id2Project.get(id).getName())
+        String names = projectId2ResourceRoleNames.keySet().stream()
```

Comment:
As you have already filtered by the project id, you could always assume the `id2Project.get(id)` will never return `null` useless JDK has bugs :), even if checking for null is a good programming practice~